File: Leetcode/35-search-insert-position.js
const searchInsert = function(nums, target) {
    
    let index = 0;
    while(nums.length !== 0)
    {
        let start = 0, last = nums.length - 1;
        middleIndex = Math.trunc((start + last) / 2);
        
        if(target > nums[middleIndex]){
             nums.splice(start, middleIndex + 1);
             start = middleIndex + 1;
             index += start;
        }
        else if(target < nums[middleIndex])
        {
            nums.splice(middleIndex);
            last = middleIndex - 1;
        }
        else return Math.trunc(index + last / 2);
    }

    return index;
};

console.log(searchInsert([6,7,8], 7));

File: Leetcode/35-Search-Insert-Position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        start = 0
        end = len(nums) - 1
        
        while start <= end:
            mid = (start + end) // 2
            if nums[mid] < target:
                start = mid + 1
            elif nums[mid] > target:
                end = mid - 1
            else:
                return mid
        
        if start >= len(nums): return len(nums)
        
        return start + (1 if nums[start] < target else 0)